src/praisonai-agents/tests/test_mcp_fail_closed.py

```
@@ -0,0 +1,87 @@
+    with patch.object(mcp_module, "MCPToolRunner", factory):
+        with pytest.raises(RuntimeError, match="initialization failed"):
+            MCP("/usr/bin/python fake_server.py", timeout=5)
+
+
+@pytest.mark.skipif(not mcp_module.MCP_AVAILABLE, reason="mcp package not installed")
+def test_mcp_nonempty_tools_ok():
+    """A server that lists at least one tool constructs successfully."""
+    class _FakeTool:
+        name = "get_current_time"
+        description = "Return the current time"
+        inputSchema = {"type": "object", "properties": {}, "required": []}
```

Ruff (0.16.2)

[warning] Mutable default value for class attribute

(RUF012)